my_each.rb

```diff
@@ -1,6 +1,141 @@
-module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+module Enumerable # rubocop:disable Style/ModuleLength
+  def my_each
+    array = to_a
+    i = 0
+    while i < array.size
+      return to_enum(:my_each) unless block_given?
+
+      yield array[i]
+      i += 1
+
     end
+
+    self
+  end
+
+  def my_each_with_index
+    array = to_a
+    i = 0
+    while i < array.size
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield array[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      return to_enum(:my_select) unless block_given?
+
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?(compare = nil)
+    result = true
+    my_each do |element|
+      result = if compare
+                 compare.class.eql?(Class) ? element.is_a?(compare) : element.to_s.match?(compare)
+               elsif block_given?
+                 yield element
+
+               else
+                 element ^ 1
+               end
+      break unless result
+    end
+
+    result
+  end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all? == array.all? # false
  ```
* [ ] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
  ```ruby
    array = [1,2,5,4,7]
    p array.my_all?(5) == array.all?(5) # false
  ```